### ISSUE TITLE
Allow to put `saveLinks = true` on async methods and do not ignore it for `putAllSync`

### ISSUE THREAD

**Issue by FredJul** [isar/isar#541]
Today, on the future 3.0.0 version, only the sync methods have the saveLinks attribute. I find it quite useful but I'm not sure to understand why this is not available as well for async methods.

Also, today the putAllSync method just ignore the attribute:

```
  @override
  List<int> putAllSync(List<OBJ> objects, {bool saveLinks = true}) {
    return putAllByIndexSync(null, objects);
  }
```

**Version**
 - Isar version: 3.0.0-main

**Comment by leisim** [comment]
Thanks for reporting that is indeed a typo.

Saving links is not supported for asynchronous methods because it's not possible to implement it in an efficient way.
Also it has subtle race conditions.

**Comment by leisim** [comment]
This has been fixed in `3.0.0-dev.9`
